### PR TITLE
importinto/lightning: check max row size when parsing csv to avoid OOM

### LINKED ISSUE #58590: check max row size when parsing csv
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

in https://github.com/pingcap/tidb/issues/40400, we have check whether a single field exceed the max entry size, but we also need to check this for the whole row, to avoid when the data file is corrupted or the input param doesn't match with the file, and causes the parser mistakenly read a vary long row and cause oom

### 4. What is your TiDB version? (Required)
master

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: close #58590

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)

import a file with 20 fields, each field is about 10+M, total row size is about 266M
```mysql
mysql> import into t from '/playground/lightning/single-file/long-row.csv';
ERROR 1105 (HY000): size of row cannot exceed the max value of txn-entry-size-limit
mysql>
```

- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by D3Hunter** [comment]
@lance6716

File: pkg/lightning/mydump/csv_parser.go [hunk 1]
```diff
@@ -398,7 +398,7 @@ func (parser *CSVParser) readUntil(chars *byteSet) ([]byte, byte, error) {
 	var buf []byte
 	for {
 		buf = append(buf, parser.buf...)
-		if len(buf) > LargestEntryLimit {
+		if parser.checkRowLen && parser.pos-parser.rowStartPos+int64(len(buf)) > int64(LargestEntryLimit) {
```
Comment (lance6716) [inline pkg/lightning/mydump/csv_parser.go:401]: in all cases, `parser.checkRowLen` is true when running here?

**Review by lance6716 — APPROVED** [review]

File: pkg/lightning/mydump/csv_parser.go (hunk 1 above)
Comment (D3Hunter) [reply pkg/lightning/mydump/csv_parser.go:401]: only public API check it, i.e. ReadRow/ReadColumns/ReadUntilTerminator

**Review by GMHDBJD — APPROVED** [review]

**Pull request merged**